### PR TITLE
[DCE-153] Walking Skeleton 

### PR DESCRIPTION
**Pull request by ampats**
Create scaffolding for dotnet application

Checklist:
- [ ] main branch is protected
- [x] includes PR template
- [x] github action that writes hello-world using ubuntu runner
- [x] basic dotnet web app that writes "Hello World" (do not do any other development)
- [x] contains README on how to run the application locally as an ubuntu user
- [x] should contain gitignore

### CODE REVIEW

**Comment by Kterlecki** [comment]
Great Work! Could you just look through some of the requirements for this Jira story, I believe a few are missing
- Readme
- Completed github action - remove the run
- etc. 
- you can leave it for this project, but try to build directories in src/nameOfProject/*.csproj.

File: .github/workflows/practice-basic-webapi-application.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+on:
+  repository_dispatch:
+    types: 
+      - webhook-test
+    
+  push:
+    branches: 
+      - main
+      - feature/DCE-153-walking-skeleton
```
Comment (sethpthomas91) [inline .github/workflows/practice-basic-webapi-application.yml]:
```suggestion
```

File: .github/workflows/practice-basic-webapi-application.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# This workflow will build a .NET project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-net
+
+name: Practice-basic-webapi-application Workflow
+
+on:
+  repository_dispatch:
+    types: 
+      - webhook-test
+
```
Comment (sethpthomas91) [inline .github/workflows/practice-basic-webapi-application.yml]:
```suggestion
```

File: .github/workflows/practice-basic-webapi-application.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    #[self-hosted, linux, x64, dm-dq-dynamic-runner, dotnet, ubuntu-latest]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      
+    - name: Build
+      run: dotnet build ./src/WebApi.csproj
+     
+    - name: Echo
+      run: echo ${{ github.event.client_payload.text }}
```
Comment (sethpthomas91) [inline .github/workflows/practice-basic-webapi-application.yml]:
```suggestion
    - name: Build
      run: echo "Hello, World"
```

**Review by sethpthomas91 — CHANGES_REQUESTED** [review]
Great first pass. Please take a look at the Jira story and make sure everything is covered.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+_Please replace this line with instructions on how to test your changes, a note
+on the devices and browsers this has been tested on, as well as any relevant
+images for UI changes._
+
+### UI accessibility checklist
+_If your PR includes UI changes, please utilize this checklist:_
+- [ ] Semantic HTML implemented?
+- [ ] Keyboard operability supported?
+- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
+- [ ] Color contrast tested?
```
Comment (sethpthomas91) [inline .github/PULL_REQUEST_TEMPLATE.md]: Great job using resources, but please take a look at the code before copying and pasting! Turn this into your own.

File: .github/workflows/practice-basic-webapi-application.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+name: Practice-basic-webapi-application Workflow
+
+on: push
+
+jobs:
+  webhook-test:
+    name: Webhook Test
+    runs-on: ubuntu-latest 
+    #[self-hosted, linux, x64, dm-dq-dynamic-runner, dotnet, ubuntu-latest]
```
Comment (sethpthomas91) [inline .github/workflows/practice-basic-webapi-application.yml]:
```suggestion
```

File: .github/workflows/practice-basic-webapi-application.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# This workflow will build a .NET project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-net
```
Comment (sethpthomas91) [inline .github/workflows/practice-basic-webapi-application.yml]:
```suggestion
```
Don't forget to clean up boiler plate code. Look at everything before you request a review and make sure you clean up unnecessary comments.

**Review by sethpthomas91 — CHANGES_REQUESTED** [review]
Very small changes, but after that this will be approved!

File: .github/workflows/practice-basic-webapi-application.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+name: Practice-basic-webapi-application Workflow
+
+on: push
+
+jobs:
+  webhook-test:
+    name: Webhook Test
```
Comment (sethpthomas91) [inline .github/workflows/practice-basic-webapi-application.yml]: Is this still a webhook test?

**Review by sethpthomas91 — APPROVED** [review]
Take a look at my one comment and please address that before squashing a merging this one. You are free to squash a merge though once that is done!

**Pull request merged**